### DISCUSSION TITLE
How do I add HTML to a popover?

### DISCUSSION THREAD

**Discussion by tmorgan497** [twbs/bootstrap#38749] [Q&A] [answered]
Hello all. I'm trying to setup a popover to act as a dialog for the user to enable filters on a table column (similar to excel's filter dialog). I have popovers enabled, but the HTML isn't transferring as expected. 

Here is my popover and the intended HTML contents:
```
<div id="associated_filter_popover_content" class="d-none">
    <input type="checkbox" class="form-check-input" id="cb_associations_filter_yes">Yes</input>
    <input type="checkbox" class="form-check-input" id="cb_associations_filter_no">No</input>
    <button onclick="clearfilters(this)" class="" id="clear_assocations_filter_btn">Clear</button>
</div>
<th>Associated?<a href="#" data-bs-toggle="popover" data-bs-trigger="focus" data-bs-content-id="associated_filter_popover_content"><img src="{% static 'filter.svg' %}" width="30px" height="30px"></a></th>
```

And the JS to initiate the popover:
```
$(document).ready(function() {
    const list = [].slice.call(document.querySelectorAll('[data-bs-toggle="popover"]'))
        list.map((el) => {
        let opts = {
            animation: false,
        }
        if (el.hasAttribute('data-bs-content-id')) {
            opts.content = document.getElementById(el.getAttribute('data-bs-content-id')).innerHTML;
            opts.html = true;
        }
        new bootstrap.Popover(el, opts);
    })
});
```

And a screenshot of the current result:
![image](https://github.com/twbs/bootstrap/assets/21185911/322c7be3-8b7b-44a8-9a2a-6bacfe8ad1f0)

**Comment by patkon** [comment, marked as answer]
Maybe the sanitizer kills your inputs and the button:
https://getbootstrap.com/docs/5.3/getting-started/javascript/#sanitizer

**Reply by tmorgan497** [reply]
Adding `sanitize: false` to the initialization of the popover did the trick.

```
        const list = [].slice.call(document.querySelectorAll('[data-bs-toggle="popover"]'))
            list.map((el) => {
            let opts = {
                animation: false,
                sanitize: false,
            }
            if (el.hasAttribute('data-bs-content-id')) {
                opts.content = document.getElementById(el.getAttribute('data-bs-content-id')).innerHTML;
                opts.html = true;
            }
            new bootstrap.Popover(el, opts);
        })
        ```

**Comment by tmorgan497** [comment]
When I add HTML to the popover, it creates a copy of all the sub elements. Then when I try to reference them later, it doesn't find the correct one. What's a method I could use to get around this? For example, my popover has checkboxes. If I use jquery to get the state of the checkbox (checked vs not checked), it doesn't reference the correct element since they are duplicates.
